src/components/ClassTable.vue

```diff
@@ -55,44 +55,21 @@ export default class ClassTable extends Vue {
 
   @Prop() private colorMapping!: { [id: string]: string };
 
-  parseWeek(data: string): number[] {
-    const res = data.match(/^(\d+)-(\d+)周(\((单|双)\))?$/);
-    const result: number[] = [];
-    if (res) {
-      const begin = parseInt(res[1]);
-      const end = parseInt(res[2]);
-      const step = res[3] ? 2 : 1;
-      for (let i = begin; i <= end; i += step) {
-        result.push(i);
-      }
-    } else {
-      data.split(",").forEach(d => {
-        const res = d.match(/^(\d+)周$/);
-        if (res) {
-          result.push(parseInt(res[1]));
-        }
-      });
-    }
-    return result;
-  }
-
-  parseBlock(data: string) {
-    const res = data.match(/^(\d+)-(\d+)节$/);
-    const result: number[] = [];
-    if (res) {
-      const begin = parseInt(res[1]);
-      const end = parseInt(res[2]);
-      for (let i = begin; i <= end; i += 1) {
-        result.push(i);
-      }
+  parseBin(data: number): number[] {
+    const result = [];
+    let i = 1;
+    while (data > 0) {
+      if (data % 2 == 1) result.push(i);
+      i += 1;
+      data = Math.floor(data / 2);
```

Comment:
Would you like to try `(123).toString(2).split('').map(x => Boolean(parseInt(x)))`

Reply:
Or you may change `Math.floor(data / 2)` to `data >> 1`

Reply:
I prefer latter one